verilog/formatting/token_annotator_test.cc

```diff
@@ -2676,6 +2676,168 @@ TEST(TokenAnnotatorTest, AnnotateFormattingWithContextTest) {
+      },
+      {
+          // 1 0 : -;
+          DefaultStyle,
+          {':', ":"},
+          {'-', "-"},
+          {NodeEnum::kUdpCombEntry},
+          {1, SpacingOptions::Undecided},
+      },
+      {
+          // 1 0 : -;
+          DefaultStyle,
+          {'-', "-"},
+          {';', ";"},
+          {NodeEnum::kUdpCombEntry},
+          {0, SpacingOptions::Undecided},
+      },
+      // Comments in UDP entries
+      {
+          // 1  /*comment*/ 0 : -;
+          DefaultStyle,
+          {'1', "1"},
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {NodeEnum::kUdpCombEntry},
+          {2, SpacingOptions::Undecided},
+      },
+      {
+          // 1  /*comment*/ 0 : -;
+          DefaultStyle,
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {'0', "0"},
+          {NodeEnum::kUdpCombEntry},
+          {1, SpacingOptions::Undecided},
+      },
+      {
+          // 1 0  // comment\n : -;
+          DefaultStyle,
+          {'0', "0"},
+          {yytokentype::TK_EOL_COMMENT, "// comment"},
+          {NodeEnum::kUdpCombEntry},
+          {2, SpacingOptions::MustAppend},
+      },
+      {
+          // 1  /*comment*/ 0 : -;
+          DefaultStyle,
+          {'1', "1"},
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {NodeEnum::kUdpSequenceEntry},
+          {2, SpacingOptions::Undecided},
+      },
+      {
+          // 1  /*comment*/ 0 : -;
+          DefaultStyle,
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {'0', "0"},
+          {NodeEnum::kUdpSequenceEntry},
+          {1, SpacingOptions::Undecided},
+      },
+      {
+          // 1 0  // comment\n : -;
+          DefaultStyle,
+          {'0', "0"},
+          {yytokentype::TK_EOL_COMMENT, "// comment"},
+          {NodeEnum::kUdpSequenceEntry},
+          {2, SpacingOptions::MustAppend},
+      },
+      {
+          // input  /* comment */ i;
+          DefaultStyle,
+          {yytokentype::TK_input, "input"},
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {NodeEnum::kUdpPortDeclaration},
+          {2, SpacingOptions::Undecided},
+      },
+      {
+          // input  /* comment */ i;
+          DefaultStyle,
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {yytokentype::SymbolIdentifier, "i"},
+          {NodeEnum::kUdpPortDeclaration},
+          {1, SpacingOptions::Preserve},
+      },
+      {
+          // input i  /* comment */;
+          DefaultStyle,
+          {yytokentype::SymbolIdentifier, "i"},
+          {yytokentype::TK_COMMENT_BLOCK, "/* comment */"},
+          {NodeEnum::kUdpPortDeclaration},
+          {2, SpacingOptions::Undecided},
+      },
+      {
+          // input i;  // comment\n
+          DefaultStyle,
+          {';', ";"},
+          {yytokentype::TK_EOL_COMMENT, "// comment"},
+          {NodeEnum::kUdpPortDeclaration},
+          {2, SpacingOptions::MustAppend},
+      },
```

Comment:
It occurred to me why we're having trouble merging these tests and seeing failure on our end.

The new functionality introduced in https://github.com/google/verible/pull/150 examines inter-token text for comment cases.  However, this test case struct does not provide any intertoken text field, so when the intertoken string_view is built here:

https://github.com/google/verible/blob/610fbea3bf873b4ef03b3331827bb21a99b2bc46/verilog/formatting/token_annotator.cc#L513

it is reading potentially uninitialized memory (making no assumptions about address layout).  To fix this, I suggest:

1) Add an assert check in the for loop below that cases with right token's enum != comment, e.g.
ASSERT_NE(right.format_token_enum, FTT::eol_comment) << "This test structure doesn't support cases that need to examine intertoken text.  Move the affected test case to AnnotateBreakAroundComments instead."
2) Move affected test cases with comments to:

https://github.com/google/verible/blob/610fbea3bf873b4ef03b3331827bb21a99b2bc46/verilog/formatting/token_annotator_test.cc#L2713

adding intertoken text as needed (in most cases, `""` or `" "` will do).

Sorry to have not spotted this earlier.

Reply:
Thanks for pointing that out, I added the warning and moved the test cases.